tools/target.py

```diff
@@ -53,19 +51,31 @@ def run(target, build, should_pull, **kwargs):
             != 0
         )
 
-        # If there are changes in tracked files in the docker folder when comparing with master,
-        # we want to build the image and we don't want to pull the image
-        if changed:
-            build = True
-            should_pull = False
-
-        if should_pull:
+        if pull:
             # Pull the image from dockerhub
             platform.pull(defaults.image, target)
 
-        # If user wants or needs to build it locally, do that
+        # If user wants or needs to build it locally, do that unless the build context is the same as origin/master
         if build:
-            platform.build(defaults.image, target)
+            if not changed:
+                cprint(
+                    "There aren't any changes to be made to the docker image. "
+                    "Skipping build, and using tagged image instead",
+                    "red",
+                    attrs=["bold"],
+                )
+            else:
+                platform.build(defaults.image, target)
+        # If the user hasn't selected build but they have changed the docker directory, then we warn them that the image
+        # isn't going to be built, even though the dirty diff implies they might want it to be built
+        else:
+            if changed:
```

Comment:
What is the construct that changed is initialised with (line 45). Not in your changes, just wondering.

Reply:
`changed` gets set to `true` on line 45 if the git diff returns a exit code other than 0 (this happens when your local docker directory is different to the docker directory that is on origin/master